spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClientUtils.java

```diff
@@ -138,6 +142,16 @@ static Map<String, String> serviceMetadata(String serviceId, Service service,
 			serviceMetadata.putAll(annotationMetadata);
 		}
 
+		if (metadataProps.addPorts()) {
```

Comment:
add port metadata to be computed here. We have a single method now responsible for all service metadata.